common/av_common.c

```diff
@@ -394,3 +398,22 @@ int mp_set_avopts_pos(struct mp_log *log, void *avobj, void *posargs, char **kv)
     }
     return success;
 }
+
+/**
+ * Free an AVPacket while ensuring that the buf and side_data are not freed.
+ *
+ * We have a particular pattern where we re-use a pre-allocated AVPacket to
+ * pass data to ffmpeg functions. In these cases, we retain ownership of the
+ * buf and side_data but must still be able to free the packet eventually. To
+ * avoid requiring explicit clearing of the fields after use, we clear them
+ * before the free.
```

Comment:
(I like this better but just a suggestion)
```suggestion
 * Must be used to free an AVPacket that was used with mp_set_av_packet().
 *
 * We have a particular pattern where we "borrow" buffers and set them
 * into an AVPacket to pass data to ffmpeg without extra copies.
 * This applies to buf and side_data, so this function clears them before freeing.
```